ffi/derive/src/ffi_fn.rs

```diff
@@ -61,11 +61,17 @@ pub fn gen_definition(fn_descriptor: &FnDescriptor) -> TokenStream {
     }
 }
 
-pub fn gen_fn_name(self_ty_name: Option<&Ident>, method_name: &Ident) -> Ident {
-    let self_ty_name = self_ty_name.map_or_else(Default::default, ToString::to_string);
+pub fn gen_fn_name(fn_descriptor: &FnDescriptor) -> Ident {
+    let method_name = &fn_descriptor.sig.ident;
+    let self_ty_name = fn_descriptor
+        .self_ty_name()
+        .map_or_else(Default::default, ToString::to_string);
+    let trait_name = fn_descriptor
+        .trait_name()
+        .map_or_else(Default::default, |trait_name| format!("__{trait_name}"));
 
     Ident::new(
-        &format!("{}__{}", self_ty_name, method_name),
+        &format!("{}{}__{}", self_ty_name, trait_name, method_name),
```

Comment:
```suggestion
        &format!("{}__{}__{}", self_ty_name, trait_name, method_name),
```

aren't you missing underscores?

Reply:
@Erigara The underscores aren't missing, but it's very difficult to understand that. Please format the `trait_name` without the underscores, and use the underscores uniformly in the final format statement.

Reply:
Format uniformly would require to introduce branching, because method name is `Struct[__Trait]__method`.
Another option would be to use `"{}{}{}"` as template and underscores would be added before where needed.

Reply:
Let's go with `{}{}{}`.